packages/openapi-code-generator/src/typescript/common/schema-builders/joi-schema-builder.ts

```diff
@@ -58,14 +67,20 @@ export class JoiBuilder extends AbstractSchemaBuilder<
     return `await ${schema}.validateAsync(${value})`
   }
 
+  public schemaTypeForType(type: string): string {
+    return type ? `${joi}.Schema<${type}>` : ""
+  }
+
   protected schemaFromRef(reference: Reference): ExportDefinition {
     const name = getSchemaNameFromRef(reference)
     const schemaObject = this.input.schema(reference)
 
+    // const type = this.typeBuilder.schemaObjectToType(reference)
     const value = this.fromModel(schemaObject, true)
 
     return {
       name,
+      // type: this.schemaTypeForType(type),
```

Comment:
Had planned to make all the schema types explicit instead, eg:
```
const schema: joi.Schema<SomeType> = ...
```
but this turned out to be challenging, particularly for string enums, and not worth the hassle right now.